src/AggregateRoots/AggregateRoot.php

```diff
@@ -29,6 +30,35 @@ abstract class AggregateRoot
 
     protected static bool $allowConcurrency = false;
 
+    /** @var \Spatie\EventSourcing\AggregateRoots\AggregateEntity[] */
+    private array $entities = [];
+
+    public function __construct()
+    {
+        $reflectionClass = new ReflectionClass($this);
+
+        foreach ($reflectionClass->getProperties() as $property) {
```

Comment:
Could we refactor this to collections? I think that will be nicer as we could use a few `filter` calls and at the end have a `each` where we could use a `AggregateEntity` in the callable.

Reply:
I agree. Even better would be if we made a package that integrates reflection with collections and add other useful stuff. One that I can think of is `$type->accepts($variable)` instead of manually comparing types. Anyways, that another discussion and I'll refactor it for now.

Reply:
maybe we could add a method `ensure` on our macro collections package?

It should throw an exception if there's an item in the collection that isn't of any of the types passed to `ensure`. 

Might also be PR worthy to Laravel.